llvm/lib/SYCLLowerIR/SYCLDeviceRequirements.cpp

```diff
@@ -99,6 +102,9 @@ llvm::computeDeviceRequirements(const module_split::ModuleDesc &MD) {
         assert(*Reqs.SubGroupSize == static_cast<uint32_t>(MDValue));
     }
   }
+
+  if (MultipleReqdWGSize)
+    Reqs.ReqdWorkGroupSize.reset();
```

Comment:
There should be a comment explaining this. My understanding is that strictly speaking, we expect only one value of `reqd_work_group_size` metadata because of per-optional-kernel-feature device code split that had supposed to happen before.

However, there is an exception when device code split is disabled, which causes kernels with different `reqd_work_group_size` requirements to be bundled together. I think that ideally we want to assert here that device code split is disabled and that otherwise `MultipleReqdWGSize` is `false`, but I'm not sure if we have access to that knowledge here.

Reply:
> However, there is an exception when device code split is disabled, which causes kernels with different reqd_work_group_size requirements to be bundled together. I think that ideally we want to assert here that device code split is disabled and that otherwise MultipleReqdWGSize is false, but I'm not sure if we have access to that knowledge here.

I agree, but yea, the device code split mode information is not present in this function. If we want to go that far, I think it makes sense to add it as a parameter to the function.